File: 0496-next-greater-element-i/0496-next-greater-element-i.ts
function nextGreaterElement(nums1: number[], nums2: number[]): number[] {
    const Stack:number[]=[];
    const map=new Map<number,number>();
    for(const n of nums2){
        while(Stack.length && n>Stack[Stack.length-1]){
            map.set(Stack.pop(),n);
        }
        Stack.push(n);
    }
    for(const n of Stack){
        map.set(n,-1);
    }

    const ans=nums1.map(element=>map.get(element));
    return ans;
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
         ArrayList<Integer> Stack=new ArrayList<>();
         HashMap<Integer,Integer> Map=new HashMap<>();
         int[] output=new int[nums1.length];
         for (int n : nums2){
            while(!Stack.isEmpty() && Stack.get(Stack.size()-1)< n){
                Map.put(Stack.remove(Stack.size()-1),n);
            }
            Stack.add(n);
         }
         for(int n:Stack){
            Map.put(n,-1);
         }
         for (int i=0;i<nums1.length;i++ ){
            output[i]=Map.get(nums1[i]);
         }

        return output;
    }
}